ci/travis/build_project.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+set -e
+
+cmake CMakeLists.txt
```

Comment:
let's update this a bit;
typically, for cmake builds, it's a good idea to move things into a build folder;
and since the Travis-CI builds can be used as reference [for users of this code], I'd say we do things:

```
mkdir -p build
cd build
cmake ..
make
cd ..
```

this also means, it would be a good idea to add the `build/` directory to the `.gitignore` file
when updating the `.gitignore` file, maybe also add all Cmake build artifacts (like CMake.cache)  or whatever else is generated
this helps avoid people from committing them by accident